edx_solutions_api_integration/users/views.py

```diff
@@ -1643,3 +1644,44 @@ def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
         })
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class UsersListWithEnrollment(UsersList):  # pylint: disable=too-many-ancestors
+    """
+    View to create Users and enroll them in a list of courses.  In addition to
+    the options provided by UsersList.post, this view accepts an optional
+    "courses" attribute in the request body, which is a list of course keys to
+    enroll in.
+
+    The response will be annotated with a "courses" attribute which reflects
+    the list of courses the user was successfully enrolled in.  Failure to
+    enroll in a given course will not propagate an error to the caller, it will
+    just cause that course's key to be omitted from the response.
+    """
+
+    def post(self, request):
+        """
+        POST /api/users/integration_test_users/
+        """
+        AUDIT_LOG.warning(
+            "API::Creating and enrolling user with UsersListWithEnrollment. "
+            "This should not be used in production"
+        )
+        response = super(UsersListWithEnrollment, self).post(request)
+        if response.status_code == status.HTTP_201_CREATED:
+            user = User.objects.get(username=request.data['username'])
+            response.data['courses'] = []
+            for course_key_string in request.data.get('courses', []):
+                try:
+                    course_key = CourseKey.from_string(course_key_string)
+                    CourseEnrollment.enroll(user=user, course_key=course_key, check_access=True)
+                except (InvalidKeyError, CourseEnrollmentException) as exc:
+                    AUDIT_LOG.warning(
+                        "API::Could not enroll %s in %s because of %s",
+                        user,
+                        course_key,
```

Comment:
@jcdyer  Unfortunately, if the exception gets thrown at this line:
```
course_key = CourseKey.from_string(course_key_string)
```
then you get this error when the log message should happen:
```
 File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/edx_solutions_api_integration/users/views.py", line 1682, in post
    course_key,
UnboundLocalError: local variable 'course_key' referenced before assignment
```

Could you use `course_key_string` here instead?

Reply:
Oh, that was dumb.  Equally dumb was I didn't add a test to verify it.  Resolving both issues now.

Reply:
Fixed!  Thanks @jcdyer .